client/task-list/tasks/payments/RemotePayments/components/PaymentMethod.js

```diff
@@ -55,73 +55,99 @@ export const PaymentMethod = ( {
 		return isOptionsUpdating();
 	} );
 
-	const installStep = useMemo( () => {
-		if ( ! plugins || ! plugins.length ) {
+	const pluginsToInstall = plugins.filter(
+		( m ) => ! activePlugins.includes( m )
+	);
+
+	useEffect( () => {
+		if (
+			pluginsToInstall.length ||
+			paymentGateway ||
+			isFetchingPaymentGateway
+		) {
 			return;
 		}
+		fetchGateway();
+	}, [ pluginsToInstall ] );
 
-		const pluginsToInstall = plugins.filter(
-			( m ) => ! activePlugins.includes( m )
-		);
-		const pluginNamesString = plugins
-			.map( ( pluginSlug ) => pluginNames[ pluginSlug ] )
-			.join( ' ' + __( 'and', 'woocommerce-admin' ) + ' ' );
+	// @todo This should updated to use the data store in https://github.com/woocommerce/woocommerce-admin/pull/6918
+	const fetchGateway = () => {
+		setIsFetchingPaymentGateway( true );
+		apiFetch( {
+			path: 'wc/v3/payment_gateways/' + key,
+		} ).then( async ( results ) => {
+			const { post_install_scripts: postInstallScripts } = results;
+			if ( postInstallScripts ) {
+				const scriptPromises = postInstallScripts.map( ( script ) =>
+					enqueueScript( script )
+				);
+				await Promise.all( scriptPromises );
+			}
+			setPaymentGateway( results );
+			setIsFetchingPaymentGateway( false );
+		} );
+	};
 
-		return {
-			key: 'install',
-			label: sprintf(
-				/* translators: %s = one or more plugin names joined by "and" */
-				__( 'Install %s', 'woocommerce-admin' ),
-				pluginNamesString
-			),
-			content: (
-				<Plugins
-					onComplete={ ( installedPlugins, response ) => {
-						createNoticesFromResponse( response );
-						recordEvent( 'tasklist_payment_install_method', {
-							plugins,
-						} );
+	const pluginNamesString = plugins
+		.map( ( pluginSlug ) => pluginNames[ pluginSlug ] )
+		.join( ' ' + __( 'and', 'woocommerce-admin' ) + ' ' );
 
-						if ( postInstallScript ) {
-							const script = document.createElement( 'script' );
-							script.src = postInstallScript;
-							document.body.append( script );
-						}
-					} }
-					onError={ ( errors, response ) =>
-						createNoticesFromResponse( response )
-					}
-					autoInstall
-					pluginSlugs={ plugins }
-				/>
-			),
-			isComplete: ! pluginsToInstall.length,
-		};
-	}, [ activePlugins, plugins ] );
+	const installStep =
```

Comment:
Just an optional random thought; I wonder if it would make sense to make the install step dependent on the `postInstallScript` promises completing? It might help the "jarring" transition which would occur if we allow it to proceed to the next step, and then a Slotfill takes over.

Reply:
We're talking about the `WooRemotePayment` slot, right?  Are you thinking of hiding the entire card until install is complete?  Or just showing a spinner?

I'm going to merge in this PR for now, but definitely would like to follow up on this and ease the flash of content that occurs after slot fill happens.